src/main/kotlin/no/nb/mlt/wls/domain/WLSService.kt

```diff
@@ -94,43 +86,32 @@ class WLSService(
             throw ItemNotFoundException("Some items do not exist in the database, and were unable to be picked")
         }
 
-        // TODO - Move some of this to Order
-        itemIds.map { itemId ->
-            // TODO - Get All Items function?
-            val item = getItem(itemId.hostName, itemId.hostId)!!
-
+        val itemsToPick = getItems(itemsPickedMap.keys.toList(), hostName)
+        itemsToPick.map { item ->
             val itemsInStockQuantity = item.quantity ?: 0
             val pickedItemsQuantity = itemsPickedMap[item.hostId] ?: 0
 
             // In the case of over-picking, set quantity to zero
```

Comment:
Delete this comment as we set quantity in the Item::pickItem function

Reply:
Should I move the logging to the Item domain class?

Reply:
I think logging can stay here, especially since this is more issue of operation than the model. It's not the item's fault that the quantity is fucky wucky

Reply:
Yeah you're right, especially since we already log in the item class, and technically it is something item/domain is more concerned about

Reply:
As discussed this is an operation that should happen in the model, as the item describes how it should behave, not the controller